udp/src/main/scala/org/apache/pekko/stream/connectors/udp/impl/UdpBind.scala

```diff
@@ -62,6 +63,9 @@ import scala.concurrent.{ Future, Promise }
       if (isAvailable(out)) {
         push(out, Datagram(data, sender))
       }
+    case (_, Terminated(_)) =>
+      listener = null
+      failStage(new IllegalStateException("UDP listener terminated unexpectedly"))
```

Comment:
I am wondering whether a `Runtime` exception is better suited to this than an `IllegalStateException` @raboof @pjfanning wdyt?

I am also wondering if there is an already existing better exception that could be used

Reply:
ConnectionException has a good name but its Javadoc is a bit specific.

> Signals that an error occurred while attempting to connect a socket to a remote address and port. Typically, the connection was refused remotely (e.g., no process is listening on the remote address/port).

Would a plain IOException be an option?